### ISSUE TITLE
Horizon V missing sound data

### ISSUE THREAD

**Issue by sean-gugler** [a2-4am/4cade#531]
The sound effect for getting hit by an enemy bullet changes when you reach level 5. The original data for this sound seems to be missing in the version you have.

I have a version with the correct sound intact. Even though my copy is not fully authentic -- it's a crack I've had since 1982 -- it isn't defaced by vanity messages. It's just missing the title screen during load.

Do you want me to send you this file, or do you want to wait until I disassemble it myself and figure out how to patch the one you've got?

**Comment by sean-gugler** [comment]
## More info

The compressed bitstream in [horizon v 19k file PRODOS (san inc crack).po](https://github.com/a2-4am/4cade/blob/main/res/dsk/horizon%20v%2019k%20file%20PRODOS%20(san%20inc%20crack).po) only fills memory up to $BBFF. It is missing 1024 more bytes of game data that belong in memory area $BC00-BFFF.

I was able to confirm the sound will play correctly if I run that disk then breakpoint the emulator to "bload" the missing data from my copy.

**Comment by sean-gugler** [comment]
I also verified that my copy loads identical bytes as your disk into memory regions $0800-$1FFF and $6000-$BBFF, so apart from the missing load screen, I have confidence my copy otherwise matches the same source as you have.

**Comment by peterferrie** [comment]
Thanks.  I will fix this once I find enough space to hold the ProDOS page so we can still write high scores.

**Comment by sean-gugler** [comment]
The crack I have uses $0600-$06FF for a temporary relocator ... maybe there?

**Comment by peterferrie** [comment]
> The crack I have uses $0600-$06FF for a temporary relocator ... maybe
> there?

I can't touch the screen-holes without possibly confusing the disk
driver (BOOTi or MD/T or whatever).  However, the screen looks like a
good candidate, apart from the high scores are also stored there, so I
have to be careful about that, too.

**Comment by peterferrie** [comment]
fixed in 0f70459cc3bcf054753023e81a5978db1759db4b

**Comment by sean-gugler** [comment]
Wow! Very fast work.

I tried playing the 20k disk. The sound is there, woo hoo! But the game locks up after entering a high score, presumably while trying to save the file. By comparison, the 19k disk is able to save the game fine even after playing the glitched sound at level 5. Tested on Applewin 1.30.8.0 configured as 128K Enhanced //e with Mockingboard.

There is another bug I've just found, present in both 19k and 20k disk images. Should I open a separate github issue for that one, or just keep this one thread? (After game over and saving new high score, play again. Game locks up upon entering warp tube.)

**Comment by peterferrie** [comment]
It looks like I used an intermediate version instead of the final one.
I will update again.

**Comment by peterferrie** [comment]
I can't reproduce the save hang.  Please include an AppleWin
save-state after the hang so that I can see.
I do see the warp-tube issue when running from ProDOS.  It doesn't
happen in TR, so we initialise the environment differently somehow.  I
will fix this.

**Comment by peterferrie** [comment]
Okay, the warp-tube issue is ProDOS's fault.  The floppy driver is
overwriting memory that it doesn't own.  It seems likely then that the
save hang has the same cause.  Copy the file to a hard disk image and
run from there.  If that has issues then I will fix those.

**Comment by sean-gugler** [comment]
I shake my fist at ProDos! :)

It looks like the [20k disk image](https://github.com/a2-4am/4cade/blob/0f70459cc3bcf054753023e81a5978db1759db4b/res/dsk/horizon%20v%2020k%20file%20PRODOS%20(san%20inc%20crack).po) in this git repo is still the one you described as an "intermediate version". Did you commit and push an updated one somewhere?

AppleWin does hang in the same spot if I copy all the files from that `.dsk` to a 32MB `.hdv`
 and boot as a Slot 7 hard drive. I attached a save-state as you requested, in case that's helpful: [horizon5_hang.zip](https://github.com/a2-4am/4cade/files/10122117/horizon5_hang.zip)

Much appreciate all the effort you put into this hobby, by the way. I came back to Apple II for the nostalgia of playing games, and now find myself taking them apart for fun. Your work has been a great inspiration.
